packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -829,6 +838,47 @@ class CupertinoTextField extends StatefulWidget {
   /// {@macro flutter.services.TextInputConfiguration.enableIMEPersonalizedLearning}
   final bool enableIMEPersonalizedLearning;
 
+  /// Used when a user inserts image-based content through the device keyboard
+  /// on Android only.
+  ///
+  /// The passed list of strings will determine which MIME types are allowed to
+  /// be inserted via the device keyboard
+  ///
+  /// This example shows how to limit your keyboard commits to specific file types
+  /// `TextField`.
+  ///
+  /// ```dart
+  /// final TextEditingController _controller = TextEditingController();
+  ///
+  /// @override
+  /// void dispose() {
+  ///   _controller.dispose();
+  ///   super.dispose();
+  /// }
+  ///
+  /// @override
+  /// Widget build(BuildContext context) {
+  ///   return Scaffold(
+  ///     body: Column(
+  ///       mainAxisAlignment: MainAxisAlignment.center,
+  ///       children: <Widget>[
+  ///         const Text('Here's a text field that supports inserting gif content:'),
+  ///         TextField(
+  ///           controller: _controller,
+  ///           contentCommitMimeTypes: ['image/gif', 'image/png'],
+  ///           onContentCommitted: (CommittedContent data) async {
+  ///             ...
+  ///           },
+  ///         ),
+  ///       ],
+  ///     ),
+  ///   );
+  /// }
+  /// ```
+  /// {@end-tool}
+  /// {@endtemplate}
```

Comment:
I think the `template` and `macro` uses in this PR here and elsewhere are malformed overall.  This comment here should begin with a starting {@template ...} at the top.  You can see an example of a template declaration [here](https://github.com/flutter/flutter/blob/dcce0db1526f3b4921e1d0f60d5d0b29ba090f94/packages/flutter/lib/src/widgets/editable_text.dart#L420) and a usage of it as a macro [here](https://github.com/flutter/flutter/blob/dcce0db1526f3b4921e1d0f60d5d0b29ba090f94/packages/flutter/lib/src/material/text_field.dart#L207).

Also, maybe it would make the most sense to declare the template in editable_text.dart and then reference it elsewhere as a macro, since EditableText is kind of the root of these new parameters.